src/applications/vaos/tests/e2e/va-appointment.cypress.spec.js

```diff
@@ -214,3 +214,212 @@ describe('VAOS direct schedule flow', () => {
     newApptTests.confirmationPageTest(additionalInfo);
   });
 });
+describe('VAOS direct schedule flow with a Cerner site', () => {
+  it('should submit form', () => {
+    initAppointmentListMock();
+    initVAAppointmentMock({ cernerUser: true });
+    cy.visit('health-care/schedule-view-va-appointments/appointments/');
+    cy.injectAxe();
+    cy.get('.va-modal-body button').click();
+    cy.findAllByRole('tab').should('exist');
+
+    // Start flow
+    cy.findByText('Schedule an appointment').click();
+
+    // Choose Type of Care
+    newApptTests.chooseTypeOfCareTest('Primary care');
+
+    // Choose Facility Type
+    newApptTests.chooseFacilityTypeTest(/VA medical center/);
+
+    // Choose VA Facility
+    newApptTests.chooseVAFacilityTest();
+
+    // Choose Clinic
+    newApptTests.chooseClinicTest();
+
+    // Choose preferred date
+    newApptTests.choosePreferredDateTest();
+
+    // Select time slot
+    newApptTests.selectTimeSlotTest();
+
+    // Reason for appointment
+    const additionalInfo = 'cough';
+    newApptTests.reasonForAppointmentTest(additionalInfo);
+
+    // Contact info
+    newApptTests.contactInfoTest();
+
+    // Review
+    newApptTests.reviewTest();
+
+    // Check form requestBody is as expected
+    cy.wait('@appointmentSubmission').should(xhr => {
+      const request = xhr.requestBody;
+
+      expect(request.clinic.siteCode).to.eq('983');
+      expect(request.clinic.clinicId).to.eq('455');
+      expect(request).to.have.property(
+        'desiredDate',
+        `${moment()
+          .add(1, 'month')
+          .startOf('month')
+          .add(4, 'days')
+          .startOf('day')
+          .format('YYYY-MM-DD')}T00:00:00+00:00`,
+      );
+      expect(request).to.have.property('dateTime');
+      expect(request).to.have.property(
+        'bookingNotes',
+        'Follow-up/Routine: cough',
+      );
+      expect(request).to.have.property('preferredEmail', 'veteran@gmail.com');
+    });
+    cy.wait('@appointmentPreferences').should(xhr => {
```

Comment:
Is this firing off a request? If so can we mock it?

Reply:
These aren't firing off requests, they're inspecting request data that was sent to a mocked request we set up.

For example, in this one, we aliased our mock for the request to set appointment preferences to `@appointmentPreferences` in our test helpers. Here, at the end of the form, we're waiting to make sure that request is finished, then inspecting the request data to make sure we're sending the data we're supposed to.

Reply:
excellent, thanks for the clarification!